Provider/BreadcrumbProvider.php

```diff
@@ -108,30 +110,41 @@ private function createBreadcrumbsFromRoutes(RouteCollection $routes)
         /** @var BreadcrumbCollectionInterface $breadcrumbs */
         $breadcrumbs = new $this->collectionClass();
 
-        $currentRoute = $routes->get($this->currentRoute);
-        $currentRouteName = $this->currentRoute;
+        $route = $routes->get($this->currentRouteName);
+        if (!$route) {
+            // we did not find the route of this request. play it safe
+            return $breadcrumbs;
+        }
+        $routeName = $this->currentRouteName;
 
-        if (false === $currentRoute->hasOption('breadcrumb')) {
+        if (false === $route->hasOption('breadcrumb')) {
             return $breadcrumbs;
         }
 
         do {
-            $options = $currentRoute->getOption('breadcrumb');
+            $options = $route->getOption('breadcrumb');
 
             if (null === $options || false === isset($options['label'])) {
                 throw new \LogicException(sprintf(
                     'Routes used as parent routes need to be configured as breadcrumbs themselves. Associated route: "%s"',
-                    $currentRouteName
+                    $routeName
                 ));
             }
 
-            $breadcrumbs->addBreadcrumbToStart(new $this->modelClass($options['label'], $currentRouteName));
+            $breadcrumbs->addBreadcrumbToStart(new $this->modelClass($options['label'], $routeName));
 
-            $parentRoute = isset($options['parent_route']) ? $options['parent_route'] : null;
+            $parentRouteName = isset($options['parent_route']) ? $options['parent_route'] : null;
 
-            $currentRoute = $routes->get($parentRoute);
-            $currentRouteName = $parentRoute;
-        } while (null !== $parentRoute);
+            $route = $routes->get($parentRouteName);
+            if (!$route) {
+                throw new \LogicException(sprintf(
+                    'Parent route "%s" specified on route "%s" not found',
+                    $parentRouteName,
+                    $routeName
+                ));
+            }
+            $routeName = $parentRouteName;
+        } while (null !== $parentRouteName);
```

Comment:
i guess we should immediately abort when the parent_route is not defined on a route, rather than try to fetch the route null from the route collection.lets refactor this on monday